### PR TITLE
Support for HAVING, TOP K, and LIMIT

### PR DESCRIPTION
**Pull request by joshua-oss**
These are all performed in post processing.

Addresses #287 and #30.

### CODE REVIEW

File: sdk/opendp/smartnoise/_ast/expressions/logical.py [hunk 1]
```diff
@@ -29,7 +32,56 @@ def children(self):
...
+                    try:
+                        l = int(l)
+                    except:
+                        l = float(l)
+                elif isinstance(r, float):
+                    l = float(l)
+                elif isinstance(r, datetime):
+                    l = datetime.fromisoformat(l)
+                elif isinstance(r, date):
+                    l = date.fromisoformat(l)
```
Comment (eedeleon) [inline sdk/opendp/smartnoise/_ast/expressions/logical.py:49]: The nested ifs should be a function.

if isinstance(l, str):
   l = resolve_string_type(l, r)
elif isinstance(r, str):
   r = resolve_str_type(r, l)

**Review by eedeleon — APPROVED** [review]
This feature has been asked for quite a bit.

Thanks for the contributions!

Do we need to update any md for how the code is working?

**Review by joshua-oss — COMMENTED** [review]
> This feature has been asked for quite a bit.
> 
> Thanks for the contributions!
> 
> Do we need to update any md for how the code is working?

I actually took the liberty of updating the design paper in the /papers/ folder as part of the safe-settings PR.  So the implementation now matches the paper.

File: sdk/opendp/smartnoise/_ast/expressions/logical.py (hunk 1 above)
Comment (joshua-oss) [reply sdk/opendp/smartnoise/_ast/expressions/logical.py:49]: Good call; I have updated

**Pull request merged**